tools/cp-utility/src/main.rs

```diff
@@ -42,21 +44,32 @@ struct CopyOperation {
 
 /// Parse command line arguments and transform into `CopyOperation`
 fn parse_args(args: Vec<&str>) -> io::Result<CopyOperation> {
-    if !(args.len() == 3 || args.len() == 4 && args[1].eq("-a")) {
+    if !(args.len() == 3
+        || args.len() == 4 && args[1].eq("-a")
+        || args.len() == 4 && args[1].eq("-r"))
+    {
         return Err(io::Error::new(
             io::ErrorKind::InvalidInput,
-            "Invalid parameters. Expected cp [-a] <source> <destination>",
+            "Invalid parameters. Expected cp [-a] <source> <destination> or  Expected cp [-r] <source> <destination>",
         ));
     }
 
-    if args.len() == 4 {
+    if args.len() == 4 && args[1].eq("-a") {
```

Comment:
nit: Should `args[1].eq("-a")` be nested so `args[1].eq("-r")` can be nested as well?